File: 2049-eliminate-maximum-number-of-monsters/2049-eliminate-maximum-number-of-monsters.py
class Solution:
    def eliminateMaximum(self, dist: List[int], speed: List[int]) -> int:
        n = len(dist)
        arrival = []
        for i in range(n):
            arrival.append(dist[i] / speed[i])
        arrival.sort()
        res = 0
        for i in range(n):
            if arrival[i] <= i:
                break
            res += 1
        return res
        

File: 2049-eliminate-maximum-number-of-monsters/2049-eliminate-maximum-number-of-monsters.java
class Solution {
    public int eliminateMaximum(int[] dist, int[] speed) {
        PriorityQueue<Double> pq = new PriorityQueue<>();
        for(int i=0;i<dist.length; i++){
            pq.add((double)dist[i]/speed[i]);
        }
        int res = 0;
        while(!pq.isEmpty()){
            if(pq.poll() <= res){
                break;
            }
            res++;
        }
        return res;
    }
}
